sandy/core/endf6.py

```diff
@@ -2475,3 +2475,162 @@ def covariance_info(self, nubar=True, xs=True, mubar=True, chi=True):
             'chi': run_chi if chi else False,
             }
         return cov_info
+
+    def _get_xs_cov(self, mf=[31, 33], mt=range(1, 10000),
+                    mat=range(1, 10000)):
+        """
+        Extract from endf6 file the cross section/nubar covariance matrices.
+
+        Parameters
+        ----------
+        mf : `int` or `list`, optional
+            MF number. The default is [31, 33].
+        mt : `int` or `list`, optional
+            MT number. The default is range(1, 10000).
+        mat : `int` or `list`, optional
+            MAT number. The default is range(1, 10000).
+
+        Returns
+        -------
+        `sandy.CategoryCov`
+            Covariance matrix.
+
+        Examples
+        --------
+        >>> endf6 = sandy.get_endf6_file('jeff_33','xs', 922380)
+        >>> cov = endf6._get_xs_cov()
+        >>> xs_cov = sandy.XsCov.from_endf6(endf6)
+        >>> assert (cov.data == xs_cov).all().all()
+        """
+        listmf_ = [mf] if isinstance(mf, int) else mf
+        listmt_ = [mt] if isinstance(mt, int) else mt
+        listmat_ = [mat] if isinstance(mat, int) else mat
+        tape = self.filter_by(listmf=listmf_, listmt=listmt_, listmat=listmat_)
+        data = []
+        for mat, mf, mt in tape.data:
+            sec = tape.read_section(mat, mf, mt)
+            for sub in sec["SUB"].values():
```

Comment:
If I give in input `mf=5`, this will fail

Reply:
I think we have to trust that the user knows what he or she is doing. In NJOY, if you produce an incorrect input (for example, in ERRORR with mf=31 asking it to parse mt=2), NJOY fails and it doesn't mean that it is badly programmed, but that we as a user are asking it to do something that doesn't make sense. 

However, if you want before processing, I can set an option to return nothing and print a message.